src/aind_dynamic_foraging_basic_analysis/metrics/trial_metrics.py

```diff
@@ -27,35 +31,176 @@ def compute_all_trial_metrics(nwb):
         print("You need to compute df_trials: nwb_utils.create_trials_df(nwb)")
         return
 
-    df = nwb.df_trials.copy()
+    df_trials = nwb.df_trials.copy()
 
-    df["RESPONDED"] = [x in [0, 1] for x in df["animal_response"].values]
+    df_trials["RESPONDED"] = [x in [0, 1] for x in df_trials["animal_response"].values]
     # Rolling fraction of goCues with a response
-    df["response_rate"] = (
-        df["RESPONDED"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
+    df_trials["response_rate"] = (
+        df_trials["RESPONDED"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
     )
 
     # Rolling fraction of goCues with a response
-    df["gocue_reward_rate"] = (
-        df["earned_reward"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
+    df_trials["gocue_reward_rate"] = (
+        df_trials["earned_reward"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
     )
 
     # Rolling fraction of responses with a response
-    df["RESPONSE_REWARD"] = [
-        x[0] if x[1] else np.nan for x in zip(df["earned_reward"], df["RESPONDED"])
+    df_trials["RESPONSE_REWARD"] = [
+        x[0] if x[1] else np.nan for x in zip(df_trials["earned_reward"], df_trials["RESPONDED"])
     ]
-    df["response_reward_rate"] = (
-        df["RESPONSE_REWARD"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
+    df_trials["response_reward_rate"] = (
+        df_trials["RESPONSE_REWARD"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
     )
 
     # Rolling fraction of choosing right
-    df["WENT_RIGHT"] = [x if x in [0, 1] else np.nan for x in df["animal_response"]]
-    df["choose_right_rate"] = (
-        df["WENT_RIGHT"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
+    df_trials["WENT_RIGHT"] = [x if x in [0, 1] else np.nan for x in df_trials["animal_response"]]
+    df_trials["choose_right_rate"] = (
+        df_trials["WENT_RIGHT"].rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True).mean()
+    )
+
+    # Rolling reward probability for best option
+    df_trials["IDEAL_OBSERVER_REWARD_PROB"] = df_trials[
+        ["reward_probabilityR", "reward_probabilityL"]
+    ].max(axis=1)
+    df_trials["ideal_observer_reward_rate"] = (
+        df_trials["IDEAL_OBSERVER_REWARD_PROB"]
+        .rolling(WIN_DUR, min_periods=MIN_EVENTS, center=True)
+        .mean()
     )
 
+    # Rolling reward probability for best option with baiting
+    if "bait_left" in df_trials:
+        df_trials["IDEAL_OBSERVER_REWARD_PROB_WITH_BAITING"] = [
+            1 if (x[0] or x[1]) else x[2]
+            for x in zip(
+                df_trials["bait_left"],
+                df_trials["bait_right"],
+                df_trials["IDEAL_OBSERVER_REWARD_PROB"],
+            )
+        ]
```

Comment:
In the `IDEAL_OBSERVER_REWARD_PROB` above, your assumption is that the ideal observer knows _which side has the greater rew_prob_, but doesn't know _the exact outcome of the randomly generated reward on each trial_. This is the same as what I defined as ["optimal-aver" for non-baiting task](https://github.com/AllenNeuralDynamics/aind-dynamic-foraging-basic-analysis/blob/a869e9e4b98d9ca16972a05e55ec49bc180151aa/src/aind_dynamic_foraging_basic_analysis/foraging_efficiency.py#L115). 

But I don't quite understand your `IDEAL_OBSERVER_REWARD_PROB_WITH_BAITING`. Basically if there was an actual reward baited on either side, the PROB = 1 (like an oracle who knows exactly where the reward is), but if no reward was baited on that trial, PROB = max(p_L, p_R)? It seems to be a weird combination of "knowing the p_reward" and "knowing the actual reward"?

Maybe using [this](https://github.com/AllenNeuralDynamics/aind-dynamic-foraging-basic-analysis/blob/a869e9e4b98d9ca16972a05e55ec49bc180151aa/src/aind_dynamic_foraging_basic_analysis/foraging_efficiency.py#L135-L146) would be better, where I assumed that the ideal observer knows the reward refilling probabilities and knows how to take advantage of the baiting dynamics (by making the "fix-and-sample" choices), but doesn't know anything about the exact baited reward.  @rachelstephlee 

Also, I think a non-baiting task doesn't need `IDEAL_OBSERVER_REWARD_PROB_WITH_BAITING`, right?

Reply:
Yeah, those were half-baked ideas I was exploring. Should I remove them from this PR, which is mostly about bias?

Reply:
Done